src/uimports.cpp

```diff
@@ -147,6 +147,20 @@ static BOOL UPDATE_IMPORTS_XX(HANDLE hProcess,
     PIMAGE_IMPORT_DESCRIPTOR piid = (PIMAGE_IMPORT_DESCRIPTOR)pbNew;
     DWORD_XX *pt;
 
+#ifdef _WIN64
+    // The offset from pbModule to any replacement import must fit into 32 bits.
+    // For simplicity, we check that the offset to the last byte fits into 32 bits,
+    // instead of the largest offset we'll actually use. The values are very similar.
+    const size_t GB4 = ((((size_t)1) << 32) - 1);
+    const size_t maxPossibleOffset = (size_t)(pbNewIid + cbNew - 1 - pbModule);
```

Comment:
This is assuming that pbNewIiid >= pbModule.

Reply:
It is supposed to be. I'd have to read more surrounding code to be sure but it is certainly supposed to be.

Reply:
I agree that the check needs to be updated.   We certainly need to have a check on x64 to avoid random crashs, as you point out.